junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/DiscoverySelectors.java

```diff
@@ -362,9 +363,48 @@ public static ClasspathResourceSelector selectClasspathResource(String classpath
 	 * @see #selectClasspathResource(String)
 	 * @see ClasspathResourceSelector
 	 * @see ReflectionSupport#tryToGetResources(String)
+	 * @deprecated Please use {@link #selectClasspathResourceByName(Set)} instead.
 	 */
-	@API(status = MAINTAINED, since = "1.13.3")
-	public static ClasspathResourceSelector selectClasspathResource(Set<Resource> classpathResources) {
+	@API(status = DEPRECATED, since = "6.0")
+	@Deprecated(since = "6.0")
+	@SuppressWarnings("removal")
+	public static ClasspathResourceSelector selectClasspathResource(
+			Set<org.junit.platform.commons.support.Resource> classpathResources) {
+		return selectClasspathResourceByName(classpathResources);
+	}
+
+	/**
+	 * Create a {@code ClasspathResourceSelector} for the supplied classpath
+	 * resources.
+	 *
+	 * <p>Since {@linkplain org.junit.platform.engine.TestEngine engines} are not
+	 * expected to modify the classpath, the supplied resource must be on the
+	 * classpath of the
+	 * {@linkplain Thread#getContextClassLoader() context class loader} of the
+	 * {@linkplain Thread thread} that uses the resulting selector.
+	 *
+	 * <p>Note: Since Java 9, all resources are on the module path. Either in
+	 * named or unnamed modules. These resources are also considered to be
+	 * classpath resources.
+	 *
+	 * <p>The {@link Set} supplied to this method should have a reliable iteration
+	 * order to support reliable discovery and execution order. It is therefore
+	 * recommended that the set be a {@link java.util.SequencedSet} (on Java 21
+	 * or higher), {@link java.util.SortedSet}, {@link java.util.LinkedHashSet},
+	 * or similar. Note that {@link Set#of(Object[])} and related {@code Set.of()}
+	 * methods do not guarantee a reliable iteration order.
+	 *
+	 * @param classpathResources a set of classpath resources; never
+	 * {@code null} or empty. All resources must have the same name, may not
+	 * be {@code null} or blank.
+	 * @since 6.0
+	 * @see #selectClasspathResource(String, FilePosition)
+	 * @see #selectClasspathResource(String)
+	 * @see ClasspathResourceSelector
+	 * @see org.junit.platform.commons.support.ResourceSupport#tryToGetResources(String)
+	 */
+	@API(status = MAINTAINED, since = "6.0")
+	public static ClasspathResourceSelector selectClasspathResourceByName(Set<? extends Resource> classpathResources) {
```

Comment:
I'm not really happy with this name. Maybe we can remove this method in the context of #4873 and add `selectClasspathResources(List<Resource>)` instead that would group by name, create a `ClasspathResourceSelector` for each group, and return a `List<ClasspathResourceSelector>`.